lib/internal/util/inspect.js

```diff
@@ -780,10 +780,20 @@ function getPrefix(constructor, tag, fallback, size = '') {
     return `[${fallback}${size}: null prototype] `;
   }
 
-  if (tag !== '' && constructor !== tag) {
-    return `${constructor}${size} [${tag}] `;
+  let result = `${constructor}${size} `;
+  if (tag !== '') {
+    const position = constructor.indexOf(tag);
+    if (position === -1) {
+      result += `[${tag}] `;
+    } else {
+      const endPos = position + tag.length;
+      if (endPos !== constructor.length &&
+        constructor[endPos + 1] === constructor[endPos + 1].toLowerCase()) {
```

Comment:
I had an off by one error writing this. It should be
```suggestion
        constructor[endPos] === constructor[endPos].toLowerCase()) {
```